torch/_utils_internal.py

```diff
@@ -155,6 +155,98 @@ def export_api_rollout_check() -> bool:
     return False
 
 
+class JustknobsConfig:
+    """Represents a lazily loaded config
+
+    This is designed to be used to specify a value in a config.
+
+    i.e. foo.bar = JustknobsConfig("//foo:bar", env_name="FORCE_FOO_BAR")
+
+    This will lazily evaluate it's configuration when compared
+    i.e. if foo.bar:
+
+    It can also be assigned to - i.e.
+    foo.bar = JustknobsConfig("//foo:bar")
+    foo.bar = True
+
+    However, this will force it to always be the overriden value. There isn't a easy way to fix this within
+    python, given how _config.py is generally used.
+
+    Note that this does allow for no JK name (so that you can use this to replace old configurations).
+    """
+
+    def __init__(self, name: Optional[str] = None, env_name=None, default: bool = True):
+        self.name = name
+        self.env_name = env_name
+        self.default = default
+
+    def __bool__(self):
+        return justknobs_feature(
```

Comment:
If I understand correctly, this means you will always reread the JK every time the config is read. It be better to read it once and then consistently use that setting through the rest of time.

Reply:
rereading JKs is explicitly part of the contract. However, just cached it for now, easy to undo in the future.